test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Execution/TestExecutionManagerTests.cs

```diff
@@ -374,6 +379,348 @@ public void SendTestResultsShouldFillInDataRowIndexIfTestIsDataDriven()
+            }
+            finally
+            {
+                DummyTestClassWithDoNotParallelizeMethods.Cleanup();
+            }
+        }
+
+        [TestMethodV1]
+        public void RunTestsForTestShouldPreferParallelSettingsFromRunSettingsOverAssemblyLevelAttributes()
+        {
+            var testCase1 = this.GetTestCase(typeof(DummyTestClassForParallelize), "TestMethod1");
+            var testCase2 = this.GetTestCase(typeof(DummyTestClassForParallelize), "TestMethod2");
+
+            TestCase[] tests = new[] { testCase1, testCase2 };
+            this.runContext.MockRunSettings.Setup(rs => rs.SettingsXml).Returns(
+                                         @"<RunSettings> 
+                                              <MSTest>
+                                                 <Parallelize>
+                                                   <Workers>2</Workers>
+                                                   <Scope>MethodLevel</Scope>
+                                                 </Parallelize>
+                                              </MSTest>
+                                            </RunSettings>");
+
+            try
+            {
+                MSTestSettings.PopulateSettings(this.runContext);
+                var testablePlatformService = this.SetupTestablePlatformService();
+                testablePlatformService.SetupMockReflectionOperations();
+
+                var originalReflectionOperation = new ReflectionOperations();
+
+                testablePlatformService.MockReflectionOperations.Setup(
+                    ro => ro.GetCustomAttributes(It.IsAny<Assembly>(), It.IsAny<Type>())).
+                    Returns((Assembly asm, Type type) =>
+                    {
+                        if (type.FullName.Equals(typeof(UTF.ParallelizeAttribute).FullName))
+                        {
+                            return new object[] { new UTF.ParallelizeAttribute { Workers = 1 } };
+                        }
+
+                        return originalReflectionOperation.GetCustomAttributes(asm, type);
+                    });
+
+                testablePlatformService.MockReflectionOperations.Setup(
+                    ro => ro.GetCustomAttributes(It.IsAny<MemberInfo>(), It.IsAny<bool>())).
+                    Returns((MemberInfo memberInfo, bool inherit) =>
+                    {
+                        return originalReflectionOperation.GetCustomAttributes(memberInfo, inherit);
+                    });
+
+                this.TestExecutionManager.RunTests(tests, this.runContext, this.frameworkHandle, new TestRunCancellationToken());
+
+                Assert.AreEqual(2, DummyTestClassForParallelize.ThreadIds.Count);
+            }
+            finally
+            {
+                DummyTestClassForParallelize.Cleanup();
+            }
+        }
+
+        [TestMethodV1]
+        [Ignore]
+        public void RunTestsForTestShouldRunTestsInTheParentDomainsApartmentState()
```

Comment:
Add tests for parallel enabled with test case filter.

Reply:
I could but that would not improve code coverage. We do not do anything special for parallel. That test would be for filtering only. I think we can take that as part of filling in the test gaps.